### PR TITLE
feat(api): Add total number of trials

### LINKED ISSUE #499: [feature] Add Trials in Experiment Spec
https://github.com/kubeflow/katib/blob/master/pkg/api/operators/apis/experiment/v1alpha2/experiment_types.go

We have TrialsSucceeded, TrialsFailed and other fields, but we do not have a field for the total number of the trials. We need it.

### PR DESCRIPTION
**Pull request by gaocegege**
Fix #499 

Signed-off-by: Ce Gao <gaoce@caicloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/501)

### CODE REVIEW

File: pkg/api/operators/apis/experiment/v1alpha2/experiment_types.go [hunk 1]
```diff
@@ -79,6 +79,9 @@ type ExperimentStatus struct {
 	// Current optimal trial parameters and observations.
 	CurrentOptimalTrial OptimalTrial `json:"currentOptimalTrial,omitempty"`
 
+	// Trials is the total number of trials owned by the experiment.
+	Trials int `json:"trials,omitempty"`
```
Comment (andreyvelich) [inline pkg/api/operators/apis/experiment/v1alpha2/experiment_types.go]: What do you think about `TotalTrials` name?

Comment (gaocegege) [reply pkg/api/operators/apis/experiment/v1alpha2/experiment_types.go]: I think in k8s community, we prefer trials. See also https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L185 and https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L383

Comment (andreyvelich) [reply pkg/api/operators/apis/experiment/v1alpha2/experiment_types.go]: Ok

**Comment by gaocegege** [comment]
Weird, https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_katib/501/kubeflow-katib-presubmit/1128140356542533632/ shows succeeded, but the bot told me failed.

**Comment by gaocegege** [comment]
Rebased, PTAL

**Pull request merged**